src/Entity/AbstractToken.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace ZfrOAuth2\Server\Doctrine\Entity;
+
+use DateTime;
+
+/**
+ * Provide basic functionality for both access tokens, refresh tokens and authorization codes
+ *
+ * Please note that scopes are stored as a saved as a string instead using associations to scope entities, mainly
+ * for performance reasons and to avoid useless database calls
+ *
+ * @author  Michaël Gallego <mic.gallego@gmail.com>
+ * @licence MIT
+ */
+abstract class AbstractToken
```

Comment:
Would surprise me as they have private constructors

> Op 31 jan. 2016 om 00:25 heeft Michaël Gallego <notifications@github.com> het volgende geschreven:
> 
> In src/Entity/AbstractToken.php:
> 
> > + */
> > +
> > +namespace ZfrOAuth2\Server\Doctrine\Entity;
> > +
> > +use DateTime;
> > +
> > +/**
> > + * Provide basic functionality for both access tokens, refresh tokens and authorization codes
> > + *
> > + * Please note that scopes are stored as a saved as a string instead using associations to scope entities, mainly
> > + * for performance reasons and to avoid useless database calls
> > + *
> > + * @author  Michaël Gallego <mic.gallego@gmail.com>
> > + * @licence MIT
> > + */
> > +abstract class AbstractToken
> Mmhh I'm wondering if it's a good idea to have entities here considering that we laready have models in the other package. @danizord, can't the model on base package be used for the entities?
> 
> —
> Reply to this email directly or view it on GitHub.
>